_posts/2022-08-22-fan-out-and-back-in-using-functions.md

```diff
@@ -0,0 +1,329 @@
+---
+title: "How to process large amounts of data with OpenFaaS functions"
```

Comment:
This title isn't doing it for me.

I think what we've made is "MapReduce" but in openfaas - https://en.wikipedia.org/wiki/MapReduce

Why do you need openfaas? For the reason that you said "to process large amounts of data"

But is that really specific and interesting enough as a title? Would someone Google that if they had this problem?

There's an element of data science, big data, and ML involved here.

Reply:
I like all of these three titles more:

https://aws.amazon.com/blogs/compute/messaging-fanout-pattern-for-serverless-architectures-using-amazon-sns/
https://theburningmonk.com/2018/04/how-to-do-fan-out-and-fan-in-with-aws-lambda/
https://dev.to/byrro/crash-course-on-fan-out-fan-in-with-aws-lambda-47g0

The diagrams on the last one are pretty close to what I had in mind when we spoke on Friday.

How to fan out and back in again with OpenFaaS
Exploring the Fan out and Fan in pattern with OpenFaaS

For a description of < 150 chars?

```
echo -n "Learn how to fan out requests to OpenFaaS functions to process them in parallel at scale, before consolidating the results by fanning back in again." |wc
      0      25     148
```